Tools/autotest/arduplane.py

```diff
@@ -681,6 +681,110 @@ def test_rc_option_camera_trigger(self):
+        print("%s" % str(m))
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        self.progress("Testing receiver enabled")
+        if (not (m.onboard_control_sensors_enabled & receiver_bit)):
+            raise NotAchievedException("Receiver not enabled")
+        self.progress("Testing receiver present")
+        if (not (m.onboard_control_sensors_present & receiver_bit)):
+            raise NotAchievedException("Receiver not present")
+        self.progress("Testing receiver health")
+        if (m.onboard_control_sensors_health & receiver_bit):
+            raise NotAchievedException()
+        self.set_parameter("SIM_RC_FAIL", 0)
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        self.progress("Testing receiver enabled")
+        if (not (m.onboard_control_sensors_enabled & receiver_bit)):
+            raise NotAchievedException("Receiver not enabled")
+        self.progress("Testing receiver present")
+        if (not (m.onboard_control_sensors_present & receiver_bit)):
+            raise NotAchievedException("Receiver not present")
+        self.progress("Testing receiver health")
+        if (not (m.onboard_control_sensors_health & receiver_bit)):
+            raise NotAchievedException("Receiver not healthy")
+        self.change_mode('MANUAL')
+
+        self.progress("Failing receiver (no-pulses)")
+        self.set_parameter("SIM_RC_FAIL", 1) # no-pulses
+        self.wait_mode('CIRCLE') # short failsafe
+        self.wait_mode('RTL') # long failsafe
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        m = self.mav.recv_match(type='SYS_STATUS', blocking=True)
+        print("%s" % str(m))
+        self.progress("Testing receiver enabled")
+        if (not (m.onboard_control_sensors_enabled & receiver_bit)):
+            raise NotAchievedException("Receiver not enabled")
+        self.progress("Testing receiver present")
+        if (not (m.onboard_control_sensors_present & receiver_bit)):
+            raise NotAchievedException("Receiver not present")
+        self.progress("Testing receiver health")
+        if (m.onboard_control_sensors_health & receiver_bit):
+            raise NotAchievedException()
```

Comment:
Same.

Reply:
Fixed.  This is all lazy code - should have a subroutine of some description or another.